src/main/duke/Duke.java

```diff
@@ -0,0 +1,391 @@
+package main.duke;
+
+import main.duke.exception.DukeException;
+import main.duke.task.Deadline;
+import main.duke.task.Event;
+import main.duke.task.Task;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class Duke {
+
+    /* Variables for Duke's use */
+    private static ArrayList<Task> taskList;
+    private static int listIndex;
+    private static Scanner scanner;
+
+    /* Constants */
+    private static final String INDENT = "\n    ";
+    private static final String H_LINE = INDENT +
+            "------------------------------------------------";
+    private static final int TASK_LIMIT = 100;
+
+
+    public static void main(String[] args) {
+
+        //Initialize necessary variables
+        init();
+
+        //Print introduction to Duke
+        introduction();
+
+        //Load in tasks if a data file exists, and show them to the user
+        loadTasksOrError();
+
+        //Re-iterate what the user types, store in list, and unmark / mark
+        while (!respondToUser()) {
+        }
+
+        //Once the user exists, save the current lists of tasks in a data file
+        saveTasksOrError();
+
+        //Salute them goodbye
+        goodBye();
+
+    }
+
+    /* Initialize variables for Duke's use */
+    private static void init() {
+        scanner = new Scanner(System.in);
+        taskList = new ArrayList<Task>();
+        listIndex = 0;
+    }
+
+    /* Print introduction to Duke */
+    private static void introduction() {
+        String logo = "     ____        _        \n"
+                + "    |  _ \\ _   _| | _____ \n"
+                + "    | | | | | | | |/ / _ \\\n"
+                + "    | |_| | |_| |   <  __/\n"
+                + "    |____/ \\__,_|_|\\_\\___|\n";
+        String introText = "\n    Hello! I'm Duke\n    What can I do for you?";
+        String introduction = logo + H_LINE + introText + H_LINE;
+        System.out.println(introduction);
+    }
+
+    /* Determine whether the user wants to create a task, exit, list, or mark */
+    private static boolean respondToUser() {
```

Comment:
This function is quite long (>30 lines), consider shortening it by breaking it up into multiple functions